dotnet/src/Connectors/Connectors.AI.Oobabooga/TextCompletion/OobaboogaTextCompletion.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.Oobabooga.TextCompletion;
+
+/// <summary>
+/// Oobabooga text completion service API.
+/// Adapted from https://github.com/oobabooga/text-generation-webui/tree/main/api-examples
+/// </summary>
+public sealed class OobaboogaTextCompletion : ITextCompletion, IDisposable
```

Comment:
We are transitioning to non-disposable connectors by ensuring that connector classes do not internally create disposable resources. Instead, if a disposable resource is needed, it should be supplied through the class constructor, indicating that the responsibility for disposal lies outside the connector. This approach enables reusing the same connector across multiple instances of the kernel and helps avoid potential issues caused by frequent creation and disposal of resources. One example is the port exhaustion due to a large number of short-lived (created and disposed per request) HTTP clients. For more information, please refer to the following link: [HttpClient Guidelines - Pooled Connections](https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#pooled-connections).

To address this specific case, please modify the class so that it no longer implements IDisposable. Instead, pass an instance of the WebSocket class as a parameter to the class.

Reply:
OK, I'll update the connector to use the fallback non disposable http client if no websocket is provided

Reply:
Guys, I will need your help to finish this up. I have no idea what's going on. 
I implemented your requested change, I ended up using a transient websocket client when no persistent one is supplied, which wasn't much of an issue (whether this is the appropriate solution can be discussed though).

But then suddently, the streaming integration test stopped working for me, without much reason for that. Oobabooga's Python websocket server would complain that the connection is closed. I tried various stuff and ended up installing Oobabooga on the same machine (I was using a remote server without any issue so far), but it didn't make it.
Upon investigating, I realized that it was the client now returning after the first (empty) packet without even closing the connection properly, whereas it used to receive 3 packets, 1st one empty, 2nd with actual content, and 3rd one to signal the end of the message.

I spent hours trying to fix that. The GetStreamingCompletionsAsync method gets out of the while(true) loop after the first packet is received, with the socket still open and no message received to end the connection.
It's as if suddenly Async won't work the same anymore.
I tried everything without any success (note that I also tried reverting any change, but it just won't come back), and for what it's worth, ChatGPT 4.0 is just as puzzled as I am.

Well maybe you won't even have the issue and something weird happened with my environment, or maybe it did work miraculously beforehand , I'm not sure what is preferable, but I'll be grateful if you can have a look.
Cheers

Reply:
@jsboige I think the problem is on our side, we merged some bad changes to streaming logic in this PR: https://github.com/microsoft/semantic-kernel/pull/1117

We have new PR open which will resolve that issue: https://github.com/microsoft/semantic-kernel/pull/1391.

As soon as it's merged, please pull this change from `main` and test again. I will ping you here as soon as fix is merged. Sorry for inconvenience.

Reply:
See the following comments. The new bug was identified and fixed in [#1bf79f0545bfd71f6291541c022e73d7ae3a78b2](https://github.com/microsoft/semantic-kernel/pull/1357/commits/1bf79f0545bfd71f6291541c022e73d7ae3a78b2)